esphome/components/mqtt/mqtt_client.cpp

```diff
@@ -347,6 +347,26 @@ void MQTTClientComponent::subscribe_json(const std::string &topic, mqtt_json_cal
   this->subscriptions_.push_back(subscription);
 }
 
+void MQTTClientComponent::unsubscribe(const std::string &topic) {
+  uint16_t ret = this->mqtt_client_.unsubscribe(topic.c_str());
+  yield();
+  if (ret != 0) {
+    ESP_LOGV(TAG, "unsubscribe(topic='%s')", topic.c_str());
+  } else {
+    delay(5);
+    ESP_LOGV(TAG, "Unsubscribe failed for topic='%s'. Will retry later.", topic.c_str());
```

Comment:
```suggestion
    ESP_LOGV(TAG, "Unsubscribe failed for topic='%s'.", topic.c_str());
```

Whoops, just saw this now. This code does not retry unsubscribing - I guess that's fine but it should not print that it does.